src/rules/promiseFunctionAsyncRule.ts

```diff
@@ -19,6 +19,28 @@ import { hasModifier } from "tsutils";
 import * as ts from "typescript";
 import * as Lint from "../index";
 
+const OPTION_FUNCTION_DECLARATION = "check-function-declaration";
+const OPTION_FUNCTION_EXPRESSION = "check-function-expression";
+const OPTION_ARROW_FUNCTION = "check-arrow-function";
+const OPTION_METHOD_DECLARATION = "check-method-declaration";
+
+const KIND_FOR_OPTION: { [arg: string]: number } = {
+    [OPTION_FUNCTION_DECLARATION]: ts.SyntaxKind.FunctionDeclaration,
+    [OPTION_FUNCTION_EXPRESSION]: ts.SyntaxKind.FunctionExpression,
+    [OPTION_ARROW_FUNCTION]: ts.SyntaxKind.ArrowFunction,
+    [OPTION_METHOD_DECLARATION]: ts.SyntaxKind.MethodDeclaration,
+};
+
+type EnabledSyntaxKinds = ReadonlySet<number>;
+
+function parseOptions(ruleArguments: string[]): EnabledSyntaxKinds {
```

Comment:
i think if this rule is enabled we should default to all kinds. Or else what's the point of turning the rule on in the first place?